packages/next-admin/src/context/FormContext.tsx

```diff
@@ -39,6 +40,19 @@ type RelationState = {
 export const FormProvider = ({ children, initialValue }: Props) => {
   const [formData, setFormData] = useState(initialValue);
   const [relationState, setRelationState] = useState<RelationState>({});
+  useEffect(() => {
+    const isDirty = !isEqual(initialValue, formData);
+    const onBeforeUnload = (e: any) => {
+      e.preventDefault();
+      e.returnValue = true
```

Comment:
I'd put an explicit message instead of letting the browser handling the message displayed in the dialog